test/kvao/_helpers.js

```diff
@@ -2,8 +2,14 @@
 
 var Promise = require('bluebird');
 
-exports.givenCacheItem = function(dataSourceFactory) {
+exports.givenCacheItem = function(dataSourceFactory, excludeFromConnector) {
   var dataSource = dataSourceFactory();
+
+  if (excludeFromConnector) {
+    // `delete connector.deleteAll` doesn't work
+    dataSource.connector[excludeFromConnector] = undefined;
```

Comment:
This is not the way how we force a datasource to use fall-back (unoptimized) methods.

See `memory` connector tests for the correct solution: https://github.com/strongloop/loopback-datasource-juggler/blob/66c54a96467b39609fec07f41b6d8018c5fc3819/test/memory.test.js#L887-L906.

The idea is to run the shared test suite twice for the built-in `kv-memory` connector. Once we run against the "full" connector, the second time we delete optimized functions from the connector instance after the datasource was created.

Th change in this file should be reverted.